infinispan-remote/src/main/java/org/hibernate/ogm/datastore/infinispanremote/InfinispanRemoteDialect.java

```diff
@@ -358,9 +357,10 @@ public Association getAssociation(AssociationKey key, AssociationContext associa
 		final String cacheName = cacheName( key );
 		ProtostreamAssociationMappingAdapter mapper = provider.getCollectionsDataMapper( cacheName );
 		return mapper.withinCacheEncodingContext( c -> {
-			QueryFactory queryFactory = Search.getQueryFactory( c );
+			QueryBuilder qb = Search.getQueryFactory( c )
+					.from( provider.getProtobufPackageName() + "." + c.getName() );
```

Comment:
This `provider.getProtobufPackageName() + "." + c.getName()` should probably be in a utility method because you are using it also in the TestHelper

Reply:
Davide, good idea. I've just added a commit with this refactor-step.